internal/controller/nonadminbackup_controller.go

```diff
@@ -70,170 +62,47 @@ const (
-		updatedNab, errBackupUpdate := function.UpdateNonAdminBackupFromVeleroBackup(ctx, r.Client, logger, &nab, &veleroBackup)
-		// Regardless if the status was updated or not, we should not
-		// requeue here as it was only status update.
-		if errBackupUpdate != nil {
-			return ctrl.Result{}, errBackupUpdate
-		} else if updatedNab {
-			logger.V(1).Info("NonAdminBackup CR - Rqueue after Status Update")
-			return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, nil
-		}
-		return ctrl.Result{}, nil
+	reconcileExit, reconcileRequeue, reconcileErr = r.ValidateVeleroBackupSpec(ctx, rLog, &nab)
+	if reconcileRequeue {
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueTimeSeconds * time.Second}, reconcileErr
+	} else if reconcileExit || reconcileErr != nil {
+		return ctrl.Result{}, reconcileErr
 	}
 
-	// Ensure labels are set for the Backup object
-	existingLabels := veleroBackup.Labels
-	naManagedLabels := function.AddNonAdminLabels(existingLabels)
-	veleroBackup.Labels = naManagedLabels
-
-	// Ensure annotations are set for the Backup object
-	existingAnnotations := veleroBackup.Annotations
-	ownerUUID := string(nab.ObjectMeta.UID)
-	nabManagedAnnotations := function.AddNonAdminBackupAnnotations(nab.Namespace, nab.Name, ownerUUID, existingAnnotations)
-	veleroBackup.Annotations = nabManagedAnnotations
-
-	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &veleroBackup, nil)
-	if err != nil {
-		logger.Error(err, "Failed to create backup", nameField, veleroBackupName)
-		return ctrl.Result{}, err
-	}
-	logger.Info("VeleroBackup successfully created", nameField, veleroBackupName)
-
-	// Phase: Created
-	// BackupAccepted: True # do not update
-	// BackupQueued: True
-	_, errUpdate = function.UpdateNonAdminPhase(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminBackupPhaseCreated)
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set NonAdminBackup Phase: Created", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionAccepted, metav1.ConditionTrue, "Validated", "Valid Backup config")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupAccepted Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
-	}
-	_, errUpdate = function.UpdateNonAdminBackupCondition(ctx, r.Client, logger, &nab, nacv1alpha1.NonAdminConditionQueued, metav1.ConditionTrue, "BackupScheduled", "Created Velero Backup object")
-	if errUpdate != nil {
-		logger.Error(errUpdate, "Unable to set BackupQueued Condition: True", nameField, req.Name, constant.NameSpaceString, req.Namespace)
-		return ctrl.Result{}, errUpdate
+	reconcileExit, reconcileRequeue, reconcileErr = r.CreateVeleroBackupSpec(ctx, rLog, &nab)
+	if reconcileRequeue {
```

Comment:
I think @shawn-hurley commented in another thread: how many options do we have :question: if it is 3 ([false,err],[false,nil] and [true,nil] for re queue and error) we do not need `reconcileExit` on these functions, correct :question: 

and would check `reconcileErr` prior to re queue

Reply:
We have the following logic that needs to be propagated back to main Reconcile function:

 1. Exit main reconcile immediately (true, false, nil)
 2. Exit main reconcile immediately with error (true, false, error)
 3. Re-queue main reconcile (false, true, nil)
 4. Continue with the main logic (false, false, nil)

We need those 4 cases because the main reconcile may continue (4th case).

Those 4 cases from which 3 have no error can not be written using one true/false return which gives us only 2 cases.

Even if we decide to include error as the logic part I still want to have explicit flag that tells if the main reconcile is re-queued, exited or continued. The logic is already complex and for the readability we definitely want those flags to be  clearly stating what is the path of the reconcile.

Reply:
Alright, what about keeping as it is, but simplify `if`s :question: 
```go
	if reconcileExit || reconcileRequeue || reconcileErr != nil{
		return ctrl.Result{Requeue: reconcileRequeue, RequeueAfter: requeueTimeSeconds * time.Second}, reconcileErr
	}
```

Reply:
The two implementations are not the same, plus your hides the desired logic inside  merged logical expression which in my opinion is harder to follow.

Reply:
@mateusoliveira43 Can we merge with the current implementation? If you think it's wrong PRs are welcome.

Reply:
Tracked via https://github.com/migtools/oadp-non-admin/issues/62 . Closing comment.